opm/simulators/wells/StandardWell.hpp

```diff
@@ -233,11 +235,74 @@ namespace Opm
             return param_.matrix_add_well_contributions_;
         }
 
+        bool doGasLiftOptimize(
+            const WellState& well_state,
+            const Simulator& ebosSimulator,
+            DeferredLogger& deferred_logger
+        ) const;
+
+        virtual void maybeDoGasLiftOptimization (
+            const WellState& well_state,
+            const Simulator& ebosSimulator,
+            DeferredLogger& deferred_logger
+        ) const;
+
```

Comment:
What is the logic of splitting these functions? I hope we can move this ++ function to the interface to make GLIFT work for MSW.

Reply:
Do you mean the function `maybeDoGasLiftOptimization()` ? We call it here : https://github.com/OPM/opm-simulators/blob/52f6f819eeebd0e6a6bb53eb8aa59ff0f271664e/opm/simulators/wells/BlackoilWellModel_impl.hpp#L898. Since it is a call through a base class (`WellInterface`) pointer I had to make it a virtual function.